program/src/matching.rs

```diff
@@ -658,6 +786,49 @@ impl BookSide {
     pub fn is_full(&self) -> bool {
         self.free_list_len <= 1 && self.bump_index >= self.nodes.len() - 1
     }
+
+    /// When a node changes, the parents' child_lowest_expiry may need to be updated.
+    ///
+    /// This function walks up the `stack` of parents and applies the change where the
+    /// previous child's `outdated_expiry` is replaced by `new_expiry`.
+    fn update_parent_lowest_expiry(
+        &mut self,
+        stack: &[(NodeHandle, bool)],
+        mut outdated_expiry: u64,
+        mut new_expiry: u64,
+    ) {
+        for (parent_h, crit_bit) in stack.iter().rev() {
```

Comment:
nitpick: the stack been an array is misleading a bit, I was confused for a moment why we are iterating and then reversing ;)